### ISSUE TITLE
Animation lags on consecutive plays

### ISSUE THREAD

**Issue by b-onc** [airbnb/lottie-ios#355]
Hello,
To explain simply, we have an onboarding page with animation, and a text. Text is hidden, animation plays, and when animation finishes, text is shown. After 5 secs (time is not important) text is hidden, animation plays, and text is shown in the completion handler again, until user performs an action while text is shown (user interaction is disabled while animation is playing)
I'm using Timer to schedule animation to rerun.
The problem is, after each rerun, animation tries to display frame 0 (or starting frame, if set differently) increasingly. First run is smooth, second run: animation tries to display frame 0 2 times, third run: 3 times, 5th run: 6 times and so on.
I've tried starting the animation from progress 0.0, 0.001, 0.01 ; playing from beginning to end, stopping the animation before running, or creating a new LOTAnimationView object altogether. None of the solutions work, and I'm inspecting the Lottie source code for an hour. I can't even figure out who sets the frame.
Screenshot: left-first run, right: 5-6th run
![screen shot 2017-08-08 at 12 07 45](https://user-images.githubusercontent.com/770464/29067322-7aaa5016-7c32-11e7-8e44-d563bc05ce1e.png)

**Comment by buba447** [comment]
Interesting!
Are you calling animation.stop() anywhere? or setting the progress?
This might be a bug in the play system.
So the frame is set by a CABasicAnimaiton that is created when you call `Play`
Ill try to code up a similar use case to yours to see if i can repoduce.

**Comment by buba447** [comment]
Any chance you can show me a simplified code example of how you are constructing this?
Whats in the completion block of the animation? What method are you using to schedule the animation?

**Comment by b-onc** [comment]
Hello,
I'm so sorry for late reply, and so thankful for the fast response!
We have an infinite scrollview that was achieved infinity by taking snapshots of first and last page at scrollViewWillBeginDragging. We were pseudo-animating the scrollView by calling the scrollViewWillBeginDragging and setContentOffset functions right after animation.play() call. The snapshotView had the afterScreenUpdates parameter set to true, so that was causing the animation to delay increasingly everytime. We set the parameter to false and problem is solved.
The problem is deep within the Core Animation framework, but we managed to solve it for ourselves with no apparent side effects.
Let me know if there's anything not clear. Thanks for all the help!
